PIL/TiffImagePlugin.py

```diff
@@ -1737,17 +1737,28 @@ def fixOffsets(self, count, isShort=False, isLong=False):
                 self.rewriteLastLong(offset)
 
 def _save_all(im, fp, filename):
-    if not hasattr(im, "n_frames"):
+    encoderinfo = im.encoderinfo.copy()
+    encoderconfig = im.encoderconfig
+    append_images = encoderinfo.get("append_images", [])
+    if not hasattr(im, "n_frames") and not len(append_images):
         return _save(im, fp, filename)
 
```

Comment:
The way that you've setup writing multiple frames, for a single frame image, there's no difference in outcome between multiple and single saving. Sure, a few more methods are called, but they're not expensive. So I would think this check is unnecessary?